packages/common/src/router-v2.js

```diff
@@ -0,0 +1,86 @@
+const { getRoutePaths } = require('./router');
+const { Router } = require('express');
+
+/**
+ * @typedef {import('express').IRouter} IRouter
+ * @typedef {import('express').Handler} Handler
+ * @typedef {Object<string, IRouter>} RouteDict
+ * @typedef {import('./router-v2-types').HttpMethods} HttpMethods
+ * @typedef {import('./router-v2-types').RouterModule} RouterModule
+ * @typedef {{ includeRoot?: boolean, backwardsCompatibilityModeEnabled?: boolean }} Options
+ */
+
+/** @type {Array<HttpMethods>} */
+const HttpMethods = [
+	'connect',
+	'delete',
+	'get',
+	'head',
+	'options',
+	'patch',
+	'post',
+	'put',
+	'trace'
+];
+
+/** @type {(str: *) => str is HttpMethods} */
+const stringIsRecognisedExport = (str) => HttpMethods.includes(str);
+
+/**
+ * @param {string} [directory]
+ * @param {Options} [options]
+ * @returns {IRouter}
+ */
+exports.getRouter = (
+	directory = __dirname,
+	{ backwardsCompatibilityModeEnabled = false, ...options } = {
```

Comment:
I find the destructuring, naming of the param + default all on the same line confusing but can live with it 😅

Reply:
I'll see what I can negotiate with the formatter

Reply:
So what I meant was my mind finds the following simpler, but it's just me being slow, no need to change anything, it's just more lines for the same thing

```js
exports.getRouter = (
	directory = __dirname,
	options = {
		includeRoot: false,
		backwardsCompatibilityModeEnabled: false
	}
) => {
const { backwardsCompatibilityModeEnabled = false, ...routeOptions } = options;
	return getRoutePaths(directory, routeOptions )
}
```

Reply:
thanks for clarifying 👍